### PR TITLE
GAWB-1066: for small (<8MB) GCS objects, proxy through orchestration

### PR DESCRIPTION
**Pull request by davidangb**
even yet more complications when downloading/viewing GCS objects.

See https://docs.google.com/a/broadinstitute.org/document/d/1TBN7IsX9Egs5iu4l4IVXuv2UrRZiMS6TgcegZ7GVIII/edit?usp=sharing_eixpa_np&ts=57ebe797 for context and discussion. 

TL;DR: if the object is small enough (arbitrarily set at 8MB), proxy it through orchestration; this way embedded images in HTML reports work. If it's larger, attempt a signed url.
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  - Acceptance criteria exists and is met
  - Note any changes to implementation from the description
  - Add notes on what you've tested
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  - Authentication
  - Authorization
  - Encryption
  - Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
- Review cycle:
  - LR reviews
  - Rest of team may comment on PR at will
  - **LR assigns to submitter** for feedback fixes
  - Submitter rebases to develop again if necessary
  - Submitter makes further commits. DO NOT SQUASH
  - Submitter updates documentation as needed
  - Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [x] **LR** sign off
- [x] **Assign to submitter** to finalize
- [x] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Squash commits and merge to develop
- [x] **Submitter**: Delete branch after merge
- [x] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [x] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [x] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [x] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

…s html reports to work.

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpGoogleServicesDAO.scala [hunk 1]
```diff
@@ -226,24 +227,45 @@ object HttpGoogleServicesDAO extends FireCloudRequestBuilding {
...
-                      // therefore, we rely on a direct link. We accept that a direct link is vulnerable to
-                      // identity problems if the current user is signed in to multiple google identies in
-                      // the same browser profile, but this is the best we can do.
-                      // generate direct link per https://cloud.google.com/storage/docs/authentication#cookieauth
-                      log.info(s"$userStr download via direct link allowed for [$objectStr]")
-                      requestContext.redirect(getDirectDownloadUrl(bucketName, objectKey), StatusCodes.TemporaryRedirect)
+                // switch solutions based on the size of the target object. If the target object is small enough,
+                // proxy it through orchestration; this allows embedded images inside HTML reports to render correctly.
+                val objSize:Int = Try(objectResponse.entity.asString.parseJson.convertTo[ObjectMetadata].size)
+                                                    .toOption.getOrElse("-1").toInt
```
Comment (helgridly) [inline src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpGoogleServicesDAO.scala:233]: Curiosity question: what does it mean when you can't get the size?

File: src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpGoogleServicesDAO.scala [hunk 2]
```diff
@@ -226,24 +227,45 @@ object HttpGoogleServicesDAO extends FireCloudRequestBuilding {
...
+                // 8MB or under ...
+                if (objSize > 0 && objSize < 8388608) {
```
Comment (helgridly) [inline src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpGoogleServicesDAO.scala:235]: Why 8MB? Should this be config'd?

**Comment by helgridly** [comment]
A couple of questions, but seems fine. 👍

File: src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpGoogleServicesDAO.scala (hunk 1 above)
Comment (davidangb) [reply src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpGoogleServicesDAO.scala:233]: It means google changed their response - which shouldn't happen. It's pretty defensive coding, but ... you know google!

File: src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpGoogleServicesDAO.scala (hunk 2 above)
Comment (davidangb) [reply src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpGoogleServicesDAO.scala:235]: I thought about config-ing it, but declined to do the work :) If it ever changes, we'll move it to config?

Comment (davidangb) [reply src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpGoogleServicesDAO.scala:235]: And ... 8MB is an educated guess that feels safe, nothing more.

**Pull request merged**